src/webgpu/api/operation/render_pipeline/primitive_topology.spec.ts

```diff
@@ -0,0 +1,421 @@
+];
+
+function getPointTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Test points are always equal to vertex locations.
+  const testLocations: TestLocation[] = [];
+  for (const location of VertexLocations) {
+    testLocations.push({ location, color: expectedColor });
+  }
+  return testLocations;
+}
+
+function getLineTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Midpoints of 3 line segments
+  return [
+    {
+      // Line {v1, v2}
+      location: Point2D.getMidpoint(VertexLocations[0], VertexLocations[1]),
+      color: expectedColor,
+    },
+    {
+      // Line {v3, v4}
+      location: Point2D.getMidpoint(VertexLocations[2], VertexLocations[3]),
+      color: expectedColor,
+    },
+    {
+      // Line {v5, v6}
+      location: Point2D.getMidpoint(VertexLocations[4], VertexLocations[5]),
+      color: expectedColor,
+    },
+  ];
+}
+
+function getPrimitiveRestartLineTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Midpoints of 2 line segments
+  return [
+    {
+      // Line {v1, v2}
+      location: Point2D.getMidpoint(VertexLocations[0], VertexLocations[1]),
+      color: expectedColor,
+    },
+    {
+      // Line {v5, v6}
+      location: Point2D.getMidpoint(VertexLocations[4], VertexLocations[5]),
+      color: expectedColor,
+    },
+  ];
+}
+
+function getLineStripTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Midpoints of 2 line segments
+  return [
+    {
+      // Line {v2, v3}
+      location: Point2D.getMidpoint(VertexLocations[1], VertexLocations[2]),
+      color: expectedColor,
+    },
+    {
+      // Line {v4, v5}
+      location: Point2D.getMidpoint(VertexLocations[3], VertexLocations[4]),
+      color: expectedColor,
+    },
+  ];
+}
+
+function getTriangleListTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Center of two triangles
+  return [
+    {
+      // Triangle {v1, v2, v3}
+      location: Point2D.getCentroid(VertexLocations[0], VertexLocations[1], VertexLocations[2]),
+      color: expectedColor,
+    },
+    {
+      // Triangle {v4, v5, v6}
+      location: Point2D.getCentroid(VertexLocations[3], VertexLocations[4], VertexLocations[5]),
+      color: expectedColor,
+    },
+  ];
+}
+
+function getTriangleStripTestLocations(expectedColor: Uint8Array): TestLocation[] {
+  // Center of two triangles
+  return [
+    {
+      // Triangle {v2, v3, v4}
+      location: Point2D.getCentroid(VertexLocations[1], VertexLocations[2], VertexLocations[3]),
+      color: expectedColor,
+    },
+    {
+      // Triangle {v3, v4, v5}
+      location: Point2D.getCentroid(VertexLocations[2], VertexLocations[3], VertexLocations[4]),
+      color: expectedColor,
+    },
+  ];
+}
+
+class PrimitiveTopologyTest extends GPUTest {
+  makeAttachmentTexture(): GPUTexture {
+    return this.device.createTexture({
+      format: kColorFormat,
+      size: { width: kRTSize, height: kRTSize, depth: 1 },
+      sampleCount: 1,
```

Comment:
nit: this is the default value and not needed.

Reply:
Done